### PR TITLE
ENT-1501:Document temporary guest pools behavior

### PR DESCRIPTION
**Pull request by sonalidhome**

### CODE REVIEW

File: docs/candlepin/guest_pool_behaviour.md [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+
+	quantity  =  masterPoolQuantity * productMultiplier
+	quantity  =  quantity *  product_Instance_Multiplier
+	masterPoolQuantity  =  quantity	
+
+## Steps performed to create Unmapped Guest pool:
+
+Once the non-custom master pool is created, current code performs following steps to create temporary guest pool: 
+~~~
+1. Calculate virt_quanity using following formula:
```
Comment (nikosmoum) [inline docs/candlepin/guest_pool_behaviour.md]: Grammar: "quanity" should be "quantity".

File: docs/candlepin/guest_pool_behaviour.md [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+	(a) if product_virt_limit  =  unlimited, 
+		virt_quantity  =  “unlimited”
+        (b) if product_virt_limit > 0,
+    		virt_quantity  =  product_virt_limit  * master_pool_quantity
+        (c) virt_quantity  =  null
+2. If virt_quanity is null then skip temporary guest pool creation process.
```
Comment (nikosmoum) [inline docs/candlepin/guest_pool_behaviour.md]: should be "quantity"

File: docs/candlepin/guest_pool_behaviour.md [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+~~~
+1. Calculate virt_quanity using following formula:
+	(a) if product_virt_limit  =  unlimited, 
+		virt_quantity  =  “unlimited”
+        (b) if product_virt_limit > 0,
+    		virt_quantity  =  product_virt_limit  * master_pool_quantity
+        (c) virt_quantity  =  null
+2. If virt_quanity is null then skip temporary guest pool creation process. 
+3. If there is any existing bonus/derived pool in pools created with the same subscription of master pool then skip guest pool creation process. 
+4. If Candlepin is running in standalone mode or host limited attribute on product is true then create temporary guest pool by cloning master pool with virt_quanity value as pool quantity.
```
Comment (nikosmoum) [inline docs/candlepin/guest_pool_behaviour.md]: should be "quantity"

File: docs/candlepin/guest_pool_behaviour.md [hunk 4]
```diff
@@ -0,0 +1,38 @@
+---
+Title: Unmapped/Temporary Guest Pool 
+---
+
+## Pools are created in 2 modes listed below:
```
Comment (nikosmoum) [inline docs/candlepin/guest_pool_behaviour.md]: This should probably include the word "When". E.g. "When pools are created in the 2 modes listed below:", and we should add a note in addition to after the 2 bullet points to clarify one of the questions of the jira description (When temporary pools are created in relation to the consumption of a relevant subscription). 

Something like "Note: Unmapped guest pools are always created during refresh pools or manifest import operations, and their creation is irrelevant to if a hypervisor consumer has consumed 1 or more from the physical pool. " should be good.

File: docs/candlepin/guest_pool_behaviour.md (hunk 1 above)
Comment (sonalidhome) [reply docs/candlepin/guest_pool_behaviour.md]: Updated variable name.

File: docs/candlepin/guest_pool_behaviour.md (hunk 2 above)
Comment (sonalidhome) [reply docs/candlepin/guest_pool_behaviour.md]: Updated variable name.

File: docs/candlepin/guest_pool_behaviour.md (hunk 3 above)
Comment (sonalidhome) [reply docs/candlepin/guest_pool_behaviour.md]: Updated variable name.

File: docs/candlepin/guest_pool_behaviour.md (hunk 4 above)
Comment (sonalidhome) [reply docs/candlepin/guest_pool_behaviour.md]: Added a note in the pool creation section.

File: docs/candlepin/guest_pool_behaviour.md [hunk 5]
```diff
@@ -0,0 +1,38 @@
+---
+Title: Unmapped/Temporary Guest Pool 
+---
+
```
[the remark this answers is not in the document]
Comment (sonalidhome) [reply docs/candlepin/guest_pool_behaviour.md]: Added description of this document.

**Comment by sonalidhome** [comment]
Thanks for the information. Updated table of contents in _data/toc.yaml.

File: docs/candlepin/guest_pool_behaviour.md [hunk 6]
```diff
@@ -0,0 +1,45 @@
+---
+Title: Unmapped/Temporary Guest Pool
```
Comment (nikosmoum) [inline docs/candlepin/guest_pool_behaviour.md]: The `Title` keyword needs to be `title` in order for the title in the table of contents to be generated properly!

Comment (sonalidhome) [reply docs/candlepin/guest_pool_behaviour.md]: Ohh...Thanks for pointing it out. Every single keyword has it's own behavior here in documents.

**Pull request merged**